examples/src/main/java/com/mapbox/navigation/examples/core/WaypointsRerouteActivity.kt

```diff
@@ -0,0 +1,366 @@
+package com.mapbox.navigation.examples.core
+
+import android.annotation.SuppressLint
+import android.location.Location
+import android.os.Bundle
+import android.view.View
+import android.view.View.GONE
+import android.view.View.VISIBLE
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import com.google.android.material.snackbar.Snackbar
+import com.google.android.material.snackbar.Snackbar.LENGTH_SHORT
+import com.mapbox.android.core.location.LocationEngineCallback
+import com.mapbox.android.core.location.LocationEngineRequest
+import com.mapbox.android.core.location.LocationEngineResult
+import com.mapbox.api.directions.v5.DirectionsCriteria
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.geojson.Point
+import com.mapbox.mapboxsdk.camera.CameraUpdateFactory
+import com.mapbox.mapboxsdk.geometry.LatLng
+import com.mapbox.mapboxsdk.location.modes.RenderMode
+import com.mapbox.mapboxsdk.maps.MapboxMap
+import com.mapbox.mapboxsdk.maps.OnMapReadyCallback
+import com.mapbox.mapboxsdk.maps.Style
+import com.mapbox.navigation.base.internal.extensions.applyDefaultParams
+import com.mapbox.navigation.base.trip.model.RouteProgress
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.directions.session.RoutesObserver
+import com.mapbox.navigation.core.directions.session.RoutesRequestCallback
+import com.mapbox.navigation.core.reroute.RerouteController
+import com.mapbox.navigation.core.reroute.RerouteState
+import com.mapbox.navigation.core.trip.session.LocationObserver
+import com.mapbox.navigation.core.trip.session.OffRouteObserver
+import com.mapbox.navigation.core.trip.session.RouteProgressObserver
+import com.mapbox.navigation.core.trip.session.TripSessionState
+import com.mapbox.navigation.core.trip.session.TripSessionStateObserver
+import com.mapbox.navigation.examples.R
+import com.mapbox.navigation.examples.utils.Utils
+import com.mapbox.navigation.examples.utils.extensions.toPoint
+import com.mapbox.navigation.ui.camera.NavigationCamera
+import com.mapbox.navigation.ui.map.NavigationMapboxMap
+import kotlinx.android.synthetic.main.activity_waypoints_reroute_layout.btnReroute
+import kotlinx.android.synthetic.main.activity_waypoints_reroute_layout.container
+import kotlinx.android.synthetic.main.activity_waypoints_reroute_layout.mapView
+import kotlinx.android.synthetic.main.activity_waypoints_reroute_layout.startNavigation
+import timber.log.Timber
+import java.lang.ref.WeakReference
+
+/**
+ * This activity shows how to:
+ * - add waypoints to the route;
+ * - manually invoke reroute;
+ * - observe reroute events with the Navigation SDK's [RoutesObserver].
+ */
+class WaypointsRerouteActivity :
```

Comment:
could you also add simulation route option like at other screens. 
For illustration: 
https://github.com/mapbox/mapbox-navigation-android/blob/15734af0cfd9fb8300091c44561475976b393882/examples/src/main/java/com/mapbox/navigation/examples/core/SimpleMapboxNavigationKt.kt#L523-L531